packages/amplify_analytics_pinpoint/android/src/main/kotlin/com/amazonaws/amplify/amplify_analytics_pinpoint/AmplifyAnalyticsBridge.kt

```diff
@@ -42,49 +43,44 @@ class AmplifyAnalyticsBridge {
         }
 
         fun recordEvent(@NonNull arguments: Any, @NonNull result: MethodChannel.Result) {
-            val argumentsMap = arguments as HashMap<*, *>
+            val argumentsMap = arguments.asPlatformChannelMap()
```

Comment:
I think it looks like a bigger change than it is. It really just centralizes where the cast occurs to a couple functions, but it doesn't change the logic at all since the same casts were occurring anyway.